src/gluonts/itertools.py

```diff
@@ -289,6 +289,62 @@ def __iter__(self):
         return filter(self.fn, self.iterable)
 
 
+@dataclass
+class WeightedIterables:
```

Comment:
Looking at just the name, I wouldn't know what this does. In some sense this is a scheduler doing resource management over multiple queues. I wish the name would make it clear that we take multiple iterators and fuse it into a single one.